### PR TITLE
scripts: Add ftrace perfetto helper scripts

### PR DESCRIPTION
**Pull request by hodgesds**
Add a set of ftrace helper scripts for making perfetto compatible ftrace scheduler profiles. I've attached a
[perfetto profile](https://github.com/user-attachments/files/17350788/sched.CPUPROFILE) of the output (Github has dumb file name extensions, but the profile should load in perfetto). The script is a little buggy and could use some improvement, but should be a good starting point.

### CODE REVIEW

File: scripts/ftrace_trim [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python3
```
Comment (hodgesds) [inline scripts/ftrace_trim:1]: This script is used to trim ftrace output until all cores are being traced.

File: scripts/sched_ftrace.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+duration=${1:-5}
+out=${2:-"$(pwd)/sched.ftrace"}
+cd /sys/kernel/debug/tracing
+echo 1 > tracing_on
+echo 1 > events/sched/sched_switch/enable
+timeout "${duration}" cat trace > "${out}.tmp"
```
Comment (arighi) [inline scripts/sched_ftrace.sh:7]: Should we read trace_pipe maybe?

**Review by arighi — APPROVED** [review]

**Review by multics69 — APPROVED** [review]
Looks good to me. Thanks a lot! @hodgesds

**Review by likewhatevs — APPROVED** [review]
pretty cool
image

**Review by htejun — APPROVED** [review]

File: scripts/sched_ftrace.sh (hunk 2 above)
Comment (multics69) [reply scripts/sched_ftrace.sh:7]: I also think it should be trace_pipe (not trace).

Comment (hodgesds) [reply scripts/sched_ftrace.sh]: That seems to work, will update.

**Pull request merged**